OrbitGl/CaptureSerializer.cpp

```diff
@@ -35,9 +35,9 @@ using orbit_client_protos::TimerInfo;
 
 ErrorMessageOr<void> CaptureSerializer::Save(const std::string& filename) {
   // Add selected functions' exact address to sampling profiler
-  const std::shared_ptr<SamplingProfiler>& profiler = Capture::capture_data_.sampling_profiler();
+  SamplingProfiler profiler = Capture::capture_data_.GetSamplingProfiler();
```

Comment:
The only line that could change saved data in `UpdateAddressInfo` is
```
address_info->set_function_name(FunctionUtils::GetDisplayName(*function));
```
And this line change function_name only if for selected address we have address_info, function, but don't have unique_callstack (otherwise we've already called this function in `ResolveCallstacks` and updated function_name).

But I agree with @florian-kuebler : if it makes sense and we want to update saved data and/or sampling_profiler, we probably want to do it when finalizing the capture and not when saving. But it changes data only if we didn't call UpdateAddressInfo function  for this address before (that called for each frame of each unique_callstacks in `ResolveCallstacks`).

Reply:
>The only line that could change saved data in UpdateAddressInfo is
address_info->set_function_name(FunctionUtils::GetDisplayName(*function));

Agree, and this will never change anything, as for the selected functions, they must be loaded, thus 
```
  if (function != nullptr) {
    function_address = FunctionUtils::GetAbsoluteAddress(*function);
    function_name = FunctionUtils::GetDisplayName(*function);
```
will always happen. So address_info will not be modified. All other internal mappings are not used in saving afterwards. Thus, I removed that code completely.